rollup-interface/src/state_machine/stf.rs

```diff
@@ -116,22 +123,39 @@ pub trait StateTransitionFunction<Vm: Zkvm, B: BlobReaderTrait> {
     ///
     /// Applies batches of transactions to the rollup,
     /// slashing the sequencer who proposed the blob on failure.
+    /// The blobs are contained into a slot whose data is contained within the `slot_data` parameter,
+    /// this parameter is mainly used within the begin_slot hook.
     /// The concrete blob type is defined by the DA layer implementation,
     /// which is why we use a generic here instead of an associated type.
     ///
     /// Commits state changes to the database
-    fn apply_slot<'a, I>(
+    ///
+    /// For now, an error can be returned if the begin slot hook returns an error. In that case
+    /// the error raised gets propagated to that level.
+    ///
+    /// We allow type complexity for now as the type is completely derived from the trait's associated types
+    /// and the trait's associated types are already complex enough.
+    #[allow(clippy::type_complexity)]
```

Comment:
Also if `anyhow::Result` is removed, it would be good to check that this clippy exception also can be removed